From 00a74e44dc3f7656f23719aa978d035a2b33c038 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 22 Jul 2016 18:54:39 +0500
Subject: [PATCH] fs: prioritize ext4 rootfs type

This would allow us to go directly to ext4 type rootfs checks
otherwise ext3 and ext2 will be tried first. This gives a
performance benefit in our case where our RootFS is known
to be ext4.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 fs/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/Makefile b/fs/Makefile
index 5fa9520..642771a 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -62,11 +62,11 @@ obj-$(CONFIG_DLM)		+= dlm/
 # Do not add any filesystems before this line
 obj-$(CONFIG_FSCACHE)		+= fscache/
 obj-$(CONFIG_REISERFS_FS)	+= reiserfs/
+obj-$(CONFIG_EXT4_FS)		+= ext4/
 obj-$(CONFIG_EXT3_FS)		+= ext3/ # Before ext2 so root fs can be ext3
 obj-$(CONFIG_EXT2_FS)		+= ext2/
 # We place ext4 after ext2 so plain ext2 root fs's are mounted using ext2
 # unless explicitly requested by rootfstype
-obj-$(CONFIG_EXT4_FS)		+= ext4/
 obj-$(CONFIG_JBD)		+= jbd/
 obj-$(CONFIG_JBD2)		+= jbd2/
 obj-$(CONFIG_CRAMFS)		+= cramfs/
-- 
1.9.1

